fast_carpenter/summary/binned_dataframe.py

```diff
@@ -7,17 +7,30 @@
 
 
 class Collector():
-    def __init__(self, filename, dataset_col, binnings):
+    def __init__(self, filename, dataset_col, binnings, file_format):
         self.filename = filename
         self.dataset_col = dataset_col
         self.binnings = binnings
+        self.file_format = file_format
 
     def collect(self, dataset_readers_list):
         if len(dataset_readers_list) == 0:
             return
 
         output = self._prepare_output(dataset_readers_list)
-        output.to_csv(self.filename, float_format="%.17g")
+
+        valid_ext = {'xlsx': 'excel', 'h5': 'hdf', 'msg': 'msgpack', 'dta': 'stata', 'pkl': 'pickle', 'p': 'pickle'}
+        for file_type in self.file_format:
+            file_ext = file_type.pop('type', None)
```

Comment:
I feel like it might be nicer to invert the understanding of the file `type` here.  In the current form you expect it to give the file extension that will be used and then look up which `save_func` to use based on that. I think we could make it more intuitive if we swap this around: the user provides `type` to directly correspond to the pandas function without the `to_` prefix.  So they give just `csv` we understand it as `to_csv`, they give `pickle` we understand it as `to_pickle`, etc.  And then the map `valid_ext` would become inverted, so we now form the extension based on the output type, e.g. `valid_ext["excel"] = "xlsx"`, `valid_ext["pickle"]="pkl"`, and we then prepend a fullstop to get the complete extension.

Reply:
I think this limits the user on what file extensions they can save as if we only get them to specify `type` and not `ext` as pickle has multiple supported extension. Currently they only need to specify the file extension in `type` and for alot of file types (e.g. `.csv`) the corresponding function in pandas is `to_csv` so there is no need to a lookup dictionary, it's just those extension (e.g. `.xlsx`) which don't map directly to their pandas function namewise.